### PR TITLE
Add support for Feature IDs

### PR DESCRIPTION
**Pull request by rbrundritt**
NetTopologySuite features don't have an ID property. As such, this library isn't currently assigning any ID value to any features when writing tiles. This creates some limitations when using the tiles elsewhere (i.e. more difficult to use feature states).

There are two common approaches used by many as a workaround;

1. Add the ID as a value in the attribute table. This is simple and easy to use.
2. Create and extention on the Geometry class that adds set/get support for an ID value. This is a bit more involved and has a greater potential of conflict as other libraries/developers may have implemented a similar extension as a solution.

Given the simplicity of option 1, this PR adds a support for specifying which attribute should be used as an ID for the feature in the tile. This is fully backwards compatible. By default it will look for an attribute called "id" (you can pass in a different value to use). If no attribute with the specified key is found, it simply won't assign any ID. If it does find a value, it then tries to parse it as a ulong, and if successful, passes the value in as the ID on the feature.
Here is an example of using an "id" attribute in a NetTopology Feature:

```
var layer = new NetTopologySuite.IO.VectorTiles.Layer()
{
    Name = "test"
};

layer.Features.Add(new Feature(new Point(0, 0), new AttributesTable(new Dictionary<string, object>() {
    { "id", 12345 }
})));
```

The writer will automatically detect the ID and use it, no changes needed.

Here is an example of passing in a custom ID value into a NetTopology Feature:

```
var layer = new NetTopologySuite.IO.VectorTiles.Layer()
{
    Name = "test"
};

layer.Features.Add(new Feature(new Point(0, 0), new AttributesTable(new Dictionary<string, object>() {
    { "myCustomID", 12345 }
})));
```

Here is an example of telling the writer to use the custom ID:

```
NetTopologySuite.IO.VectorTiles.Mapbox.MapboxTileWriter.Write(tile, fs, null, "myCustomID");
```

### CODE REVIEW

**Comment by FObermaier** [comment]
LGTM, it'd be very nice if you'd use the code inside the nice documentation to set up a unit test.

File: src/NetTopologySuite.IO.VectorTiles.Mapbox/MapboxTileReader.cs [hunk 1]
```diff
@@ -275,26 +301,31 @@ private static (MapboxCommandType, int) ParseCommandInteger(uint commandInteger)
         private static IAttributesTable ReadAttributeTable(Tile.Feature mbTileFeature, List<string> keys, List<Tile.Value> values)
         {
             var att = new AttributesTable();
-            for (int i = 0; i < mbTileFeature.Tags.Count; i += 2)
+
+            //Ensure there are enough keys to read.
+            if (keys.Count == mbTileFeature.Tags.Count)
```
Comment (FObermaier) [inline src/NetTopologySuite.IO.VectorTiles.Mapbox/MapboxTileReader.cs:306]: Don't we have to check `keys.Count  == mbTileFeature.Tags * 2` as key and value are stored intermittend?
Do we have to check this at all?

Comment (rbrundritt) [reply src/NetTopologySuite.IO.VectorTiles.Mapbox/MapboxTileReader.cs:306]: Good call on that count comparison, I should have tested with more than one attribute. 

The reason for this test is that someone may decide they only want to read the geometries from a Mapbox tile and not the attributes. A quick and way to do this is to `tile.Layers[0].Keys.Clear();` on a Mapbox tile and the reader will not extract the attributes. That said, perhaps this is an edge case that can be done another way (clear the layers keys and each features tags). I'll remove this comparison and modify my test as needed.

Comment (rbrundritt) [reply src/NetTopologySuite.IO.VectorTiles.Mapbox/MapboxTileReader.cs]: While I have your attention, where would be a good place for me to create some documentation for this library? I was going to add some to the README or the projects WIKI but wasn't sure if there was another preferred location for this.
